http-client/src/main/java/io/airlift/http/client/HttpClientBinder.java

```diff
@@ -70,11 +78,13 @@ public HttpClientBinder bindGlobalFilter(HttpRequestFilter filter)
     {
         private final HttpClientModule module;
         private final Multibinder<HttpRequestFilter> multibinder;
+        private final Multibinder<HttpStatusListener> statusListenerBinder;
 
-        public HttpClientBindingBuilder(HttpClientModule module, Multibinder<HttpRequestFilter> multibinder)
+        public HttpClientBindingBuilder(HttpClientModule module, Multibinder<HttpRequestFilter> multibinder, Multibinder<HttpStatusListener> statusListenerBinder)
         {
-            this.module = module;
-            this.multibinder = multibinder;
+            this.module = requireNonNull(module, "module is null");
+            this.multibinder = requireNonNull(multibinder, "multibinder is null");
```

Comment:
Let's rename the field and parameter to `filterBinder` now that we have multiple multibinders (the old name was bad anyway)